src/Responder.cpp

```diff
@@ -0,0 +1,153 @@
+#include "Responder.h"
+
+void Responder::on(std::string task_name, 
+                   std::function <void(const std::stringstream&, std::stringstream&)> handler) {
+    handlers_[task_name] = handler;
+};
+
+void Responder::run(std::string task_name, const std::stringstream& input, std::stringstream& output) {
+    handlers_[task_name](input, output);
+};
+
+std::vector<std::string> Responder::listTasks() {
+    std::vector<std::string> tasks;
+    for (std::map<std::string, std::function <void(const std::stringstream&, std::stringstream&)>>::iterator it = handlers_.begin(); 
+        it != handlers_.end(); ++it) {
+            tasks.push_back(it->first);
+        }
```

Comment:
tasks.reserve(handlers_.size());
for (auto const& pair : handlers_)
  tasks.push_back(pair.first);